src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```diff
@@ -0,0 +1,755 @@
+    throw xrt_core::error("Failed to update SC flash image");
+
+  // Bring back mgmt pf
+  try {
+    dev->device_online();
+  } catch (const xrt_core::error& e) {
+    throw xrt_core::error(e.what() + std::string(" Please warm reboot."));
+  }
+
+  std::cout << boost::format("%-8s : %s \n\n") % "INFO" % "SC firmware image has been programmed successfully.";
+}
+
+// Helper function for header info
+static std::string
+file_size(const std::string& _file)
+{
+  std::ifstream in(_file, std::ifstream::ate | std::ifstream::binary);
+  auto total_size = std::to_string(in.tellg());
+  int strSize = static_cast<int>(total_size.size());
+
+  // if file size is > 3 digits, insert commas after every 3 digits
+  static const int COMMA_SPACING = 3;
+  if (strSize > COMMA_SPACING) {
+    for (int i = (strSize - COMMA_SPACING); i > 0; i -= COMMA_SPACING)
+      total_size.insert(static_cast<size_t>(i), 1, ',');
+  }
+
+  return total_size + std::string(" bytes");
+}
+
+// Helper function for header info
+// Seperates the given file path into its path
+// and name components
+static std::pair<std::string, std::string>
+deployment_path_and_filename(std::string file)
+{
+  using tokenizer = boost::tokenizer<boost::char_separator<char>>;
+  boost::char_separator<char> sep("\\/");
+  tokenizer tokens(file, sep);
+  std::string dsafile = "";
+  for (auto tok_iter = tokens.begin(); tok_iter != tokens.end(); ++tok_iter) {
+    if ((std::string(*tok_iter).find(XSABIN_FILE_SUFFIX) != std::string::npos)
+        || (std::string(*tok_iter).find(DSABIN_FILE_SUFFIX) != std::string::npos))
+      dsafile = *tok_iter;
+  }
+  auto pos = file.rfind('/') != std::string::npos ? file.rfind('/') : file.rfind('\\');
+  std::string path = file.erase(pos);
+
+  return {dsafile, path};
+}
+
+// Helper function for header info
+static std::string
+get_file_timestamp(const std::string& _file)
+{
+  boost::filesystem::path p(_file);
+  if (!boost::filesystem::exists(p))
+    throw xrt_core::error("Invalid platform path.");
+
+  std::time_t ftime = boost::filesystem::last_write_time(boost::filesystem::path(_file));
+  std::string timeStr(std::asctime(std::localtime(&ftime)));
+  timeStr.pop_back();  // Remove the new-line character that gets inserted by asctime.
+  return timeStr;
+}
+
+static void
+pretty_print_platform_info(const boost::property_tree::ptree& _ptDevice, const std::string& vbnv)
+{
+  std::cout << boost::format("%s : [%s]\n") % "Device" % _ptDevice.get<std::string>("platform.bdf");
+  std::cout << std::endl;
+  std::cout << "Current Configuration\n";
+
+  std::cout << boost::format("  %-20s : %s\n") % "Platform" % _ptDevice.get<std::string>("platform.current_shell.vbnv", "N/A");
+  std::cout << boost::format("  %-20s : %s\n") % "SC Version" % _ptDevice.get<std::string>("platform.current_shell.sc_version", "N/A");
+  std::cout << boost::format("  %-20s : %s\n") % "Platform ID" % _ptDevice.get<std::string>("platform.current_shell.id", "N/A");
+  std::cout << std::endl;
```

Comment:
Mixing and matching std::endl or "\n".  
Please use "\n" unless you wish to have the buffer flushed, then use std::endl